modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/custom/form/FormView.es.js

```diff
@@ -48,53 +49,107 @@ import {paginationReducer} from './reducers/index.es';
  */
 const useFormSubmit = ({apiRef, containerRef}) => {
 	const {activePage, pages} = useFormState();
-	const {submittable} = useConfig();
+	const {submittable, validateCsrfTokenURL} = useConfig();
 
 	const handleFormSubmitted = useCallback(
 		(event) => {
 			event.preventDefault();
 
-			apiRef.current
-				.validate()
-				.then((validForm) => {
-					if (validForm) {
-						const liferayForm =
-							event.target.id &&
-							Liferay.Form.get(event.target.id);
-
-						const validLiferayForm = !Object.keys(
-							liferayForm?.formValidator?.errors ?? {}
-						).length;
-
-						if (!validLiferayForm) {
-							return;
-						}
-
-						if (submittable) {
-							Liferay.Util.submitForm(event.target);
-						}
-
-						Liferay.fire('ddmFormValid', {
-							formWrapperId: event.target.id,
-						});
-
-						Liferay.fire('ddmFormSubmit', {
-							formId: getFormId(
-								getFormNode(containerRef.current)
-							),
-						});
+			fetch(validateCsrfTokenURL, {
+				headers: new Headers({
+					Accept: 'application/json',
+					'Content-Type': 'application/json',
+				}),
+				method: 'GET',
+			})
```

Comment:
This fetch() can be replaced by an useResource hook, however I looked into the code and it is used only when extra functionality is needed such as caching, fetching attempts, network status.

Do you think it would be good to use it only as a fetch wrapper or add these extra functionalities?